salt/metalk8s/addons/ui/config/metalk8s-shell-ui-config.yaml.j2

```
@@ -2,14 +2,25 @@
 
 {%- set dex_defaults = salt.slsutil.renderer('salt://metalk8s/addons/dex/config/dex.yaml.j2', saltenv=saltenv) %}
 {%- set dex = salt.metalk8s_service_configuration.get_service_conf('metalk8s-auth', 'metalk8s-dex-config', dex_defaults) %}
+{%- set metalk8s_ui_defaults = salt.slsutil.renderer(
+        'salt://metalk8s/addons/ui/config/metalk8s-ui-config.yaml', saltenv=saltenv
+    )
+%}
+
+{%- set metalk8s_ui_config = salt.metalk8s_service_configuration.get_service_conf(
+        'metalk8s-ui', 'metalk8s-ui-config', metalk8s_ui_defaults
+    )
+%}
+
+{%- set normalized_base_path = ('' if metalk8s_ui_config.spec.basePath.strip('/') == '' else '/') + metalk8s_ui_config.spec.basePath.strip('/') %}
```

OK for normalizing, but let's find a value that we can use everywhere below, otherwise it's a bit confusing...

For instance:

```suggestion
{%- set normalized_base_path = '/' if metalk8s_ui_config.spec.basePath == '/' else metalk8s_ui_config.spec.basePath.rstrip('/') %}
```

(so, `normalized` is `/` if `base` is `/`, and `/toto` if `base` is `/toto` or `/toto/`)

Which can be used like:

- `oidc.redirectUrl = "...:8443{{normalized}}"`
- `options.main["...:8443{{normalized}}"]`
- `activeIfMatches = "...:8443{{normalized}}($|/(?!alerts|docs).*)"`

And since all of these URLs share the same prefix, you could even define:

```jinja
{%- set metalk8s_ui_url = "https://" ~ grains.metalk8s.control_plane_ip ~
                          ":8443" ~ normalized_base_path %}
```